lib/src/rules/unnecessary_const.dart

```diff
@@ -60,13 +62,9 @@ class _Visitor extends SimpleAstVisitor {
   void visitInstanceCreationExpression(InstanceCreationExpression node) {
     if (node.keyword == null) return;
 
-    // remove keyword and check if there's const error
-    final oldKeyword = node.keyword;
-    node.keyword = null;
-    final isConstWithoutKeyword = node.isConst;
-    node.keyword = oldKeyword;
-
-    if (isConstWithoutKeyword && node.keyword?.type == Keyword.CONST) {
+    // todo (pq): remove cast when `inConstantContext` is public on Element
+    var inConstContext = (node as ExpressionImpl).inConstantContext;
+    if (inConstContext && node.keyword?.type == Keyword.CONST) {
```

Comment:
Consider moving `node.keyword?.type == Keyword.CONST` to line 63. It's a less expensive check so it should be done first, and there's really no reason to check that the keyword is non-null in t addition to checking that it is the right keyword.

Reply:
Good catch.  Konstantin had a similar suggestion. Updated.